File: 70-climbing-stairs/climbing-stairs.py
import math

class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        sqrt5 = math.sqrt(5)
        phi = (1 + sqrt5) / 2
        psi = (1 - sqrt5) / 2
        return int((phi**(n + 1) - psi**(n + 1)) / sqrt5)

File: 70-climbing-stairs/climbing-stairs.cs
using System;

public class Solution {
    public int ClimbStairs(int n) {
        double sqrt5 = Math.Sqrt(5);
        double phi = (1 + sqrt5) / 2;
        double psi = (1 - sqrt5) / 2;
        return (int)((Math.Pow(phi, n + 1) - Math.Pow(psi, n + 1)) / sqrt5);
    }
}
